stdlib/3.7/dataclasses.pyi

```diff
@@ -17,14 +17,26 @@ def asdict(obj: Any, *, dict_factory: Callable[[List[Tuple[str, Any]]], _T]) ->
 def astuple(obj: Any) -> Tuple[Any, ...]: ...
 @overload
 def astuple(obj: Any, *, tuple_factory: Callable[[List[Any]], _T]) -> _T: ...
-@overload
-def dataclass(_cls: Type[_T]) -> Type[_T]: ...
-@overload
-def dataclass(_cls: None) -> Callable[[Type[_T]], Type[_T]]: ...
-@overload
-def dataclass(
-    *, init: bool = ..., repr: bool = ..., eq: bool = ..., order: bool = ..., unsafe_hash: bool = ..., frozen: bool = ...
-) -> Callable[[Type[_T]], Type[_T]]: ...
+
+if sys.version_info >= (3, 8):
+    @overload
+    def dataclass(__cls: Type[_T]) -> Type[_T]: ...
```

Comment:
Took me a long time to figure out that the difference between the branches is `__cls` vs. `_cls`. Maybe it's worth a comment.